linesearcher_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"testing"
+)
+
+func TestMoreThuente(t *testing.T) {
+	d := 0.001
+	c := 0.001
+	ls := &MoreThuente{
+		DecreaseFactor:  d,
+		CurvatureFactor: c,
+	}
+	testLinesearcher(ls, d, c, true, t)
+}
+
+func TestBisection(t *testing.T) {
+	c := 0.001
+	ls := &Bisection{
+		GradConst: c,
+	}
+	testLinesearcher(ls, 0, c, true, t)
+}
+
+func TestBacktracking(t *testing.T) {
+	d := 0.001
+	ls := &Backtracking{
+		FuncConst: d,
+	}
+	testLinesearcher(ls, d, 0, false, t)
+}
+
+type lsProblem struct {
+	name string
+	f    func(float64) float64
+	g    func(float64) float64
+}
+
+// newPlassmannFunc returns a test function labeled as (5.3) in
```

Comment:
Shouldn't these be in functions? Even if specialized I think we/someone will find it useful to have an easily accessible set of functions. We could provide a one-D wrapper here or something.